### PR TITLE
Added a small utility method to display a warnings.

### PR DESCRIPTION
**Pull request by maxcoulombe**
Will print warning if flag is passed after positional arguments e.g. `vault <command> -a b -c`
In this example `-c` will be interpreted as an argument because [go-flag stops parsing after the first non-flag arg](https://gobyexample.com/command-line-flags).

This behaviour may cause confusion or commands to run with unexpected results.
The decision to display a warning instead of crashing was to avoid potentially breaking existing scripts and integrations.

flag_warning

Small clean-up:
- Uniformized receiver name as `f`

### CODE REVIEW

File: command/base.go [hunk 1]
```diff
@@ -582,8 +584,17 @@ func (f *FlagSets) Completions() complete.Flags {
...
+// Warnings, if any, regarding the arguments format are sent to stdout
 func (f *FlagSets) Parse(args []string) error {
-	return f.mainSet.Parse(args)
+	err := f.mainSet.Parse(args)
+
+	var out bytes.Buffer
+	printArgsWarningIfAny(&out, f.Args())
+	if out.String() != "" {
+		f.ui.Warn(out.String())
+	}
```
Comment (calvn) [inline command/base.go:595]: Alternatively, you can have `printArgsWarningIfAny` return the warning as a string, and output the string if it's non-empty.

File: command/base_helpers.go [hunk 2]
```diff
@@ -292,3 +292,16 @@ func parseFlagFile(raw string) (string, error) {
 
 	return raw, nil
 }
+
+func printArgsWarningIfAny(w io.Writer, args []string) {
```
Comment (tomhjp) [inline command/base_helpers.go:296]: This doesn't actually do any printing, how about `generateFlagWarnings`, `generateTrailingFlagWarnings` or similar?

**Review by tomhjp — COMMENTED** [review]
Looks good! Just some small suggestions

**Review by austingebauer — APPROVED** [review]
LGTM with the recent commits 👍

**Review by tomhjp — APPROVED** [review]

**Review by calvn — APPROVED** [review]

**Pull request merged**